nav2_util/include/nav2_util/node_utils.hpp

```diff
@@ -124,6 +124,70 @@ void declare_parameter_if_not_declared(
   }
 }
 
+using NodeParamInterfacePtr = rclcpp::node_interfaces::NodeParametersInterface::SharedPtr;
+using ParamDescriptor = rcl_interfaces::msg::ParameterDescriptor;
+
+/// If the parameter is already declared, returns its value,
+/// otherwise declares it and returns the default value
+/**
+ * If the parameter is already declared, returns its value,
+ * otherwise declares it and returns the default value.
+ * The method will by default print a warning when the override is missing.
+ *
+ * \param[in] logger Node logging interface
+ * \param[in] param_interface Node parameter interface
+ * \param[in] param_name Name of the parameter
+ * \param[in] default_value Default value of the parameter
+ * \param[in] warn_if_no_override If true, prints a warning whenever the parameter override is missing
+ * \param[in] param_descriptor Optional parameter descriptor
+ * \return The value of the param from the override if existent, otherwise the default value
+ */
+template<typename ParamType>
+ParamType declare_or_get_param(
```

Comment:
```suggestion
ParamType declare_or_get_parameter(
```

Maybe just rename to not be shortened